tests/test_goto.py

```diff
@@ -0,0 +1,72 @@
+from setup import TextDocumentTestCase, close_test_view
+import sublime
+from os.path import dirname
+
+TEST_FILE_PATH = dirname(__file__) + "/goto.txt"
+
+RESPONSE = [
+    {
+        'uri': TEST_FILE_PATH,
+        'range':
+        {
+            'start':
+            {
+                'character': 2,
+                'line': 0
+            },
+        }
+    }
+]
+
+
+class GotoTestCase(TextDocumentTestCase):
+
+    def setUp(self):
+        super().setUp()
+        yield 100
+        self.view.run_command('insert', {"characters": 'hello there'})
+        yield 100
+        self.goto_view = sublime.active_window().open_file(TEST_FILE_PATH)
+        yield 100
+        self.goto_view.run_command('insert', {'characters': 'foo'})
+        yield 100
+        self.view.window().focus_view(self.view)  # go back to first view
+        yield 100
+
+    def tearDown(self):
+        close_test_view(self.goto_view)
+        super().tearDown()
+
+
+class LspGotoCommandTests(GotoTestCase):
+
+    def do_common_checks(self):
+        yield 100
+        view = sublime.active_window().active_view()
+        self.checkTrue(view.file_name().endswith("goto.txt"))
+        line1, col1 = view.rowcol(view.sel()[0].a)
+        line2, col2 = view.rowcol(view.sel()[0].b)
+        self.checkEqual(line1, 0)
+        self.checkEqual(col1, 2)
+        self.checkEqual(line2, 0)
+        self.checkEqual(col2, 2)
+
+    def test_definition(self):
+        self.client.responses['textDocument/definition'] = RESPONSE
+        self.view.run_command('lsp_symbol_definition')
+        self.do_common_checks()
```

Comment:
I believe you may need to `yield self.do_common_checks()` here to turn `test_definition` into a generator (AKA an async test method). I have a feeling I've tried this before and given up on it, would be awesome if we can get it working for the DRY factor.

Reply:
You're totally right. I had made the test for textDocument/definition correctly and then decided to put the common checks into a method. They're most definitely working now with the latest changes.